docker/tif-metadata/run.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+# -*- encoding: utf-8 -*-
+"""
+Usage:
+  run.py --bucket-name=<BUCKET_NAME> --key=<KEY> [--delete-original]
+  run.py -h | --help
+
+Options:
+  -h --help              Show this message.
+  --bucket-name=<BUCKET_NAME>
+  --key=<>
+  --delete-original      Delete the original source key.
+
+"""
+
+import subprocess
+import tempfile
+import hashlib
+
+import json
+import boto3
+import docopt
+from os.path import basename
+
+
+args = docopt.docopt(__doc__)
+
+job_bucket = args["--bucket-name"]
+job_s3_key = args["--key"]
+delete_original = bool(args['--delete-original'])
+# actually this should be in the json job separate from the key
+image_bucket = "wellcomecollection-tif-derivatives"
+
+client = boto3.client("s3")
+job_filename = basename(job_s3_key)
+client.download_file(Bucket=job_bucket, Key=job_s3_key,
+                     Filename=job_filename)
+
+with open(job_filename, "r") as job_file:
+    job = json.load(job_file)
+
+tasks = job['task_list']
+print(f"Starting download of {len(tasks)} images from s3")
+processes = []
+for task in tasks:
+    _, tmp_fp = tempfile.mkstemp()
```

Comment:
You could move this into a function and then use a list comprehension to make it clearer what's going on.

i.e. 

```py
def download_source(source, processes = []):
   _, tmp_fp = tempfile.mkstemp()
  s3_source = task["source"]
  p = subprocess.Popen(["aws", "s3", "cp", s3_source, tmp_fp])
  processes.append((p, task, tmp_fp))

[download_source(task["source"], processes) for task in tasks]
```